src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -76,12 +76,15 @@
 NSString * const MTRArrayValueType = @"Array";
 
 class MTRDataValueDictionaryCallbackBridge;
+template <typename DecodableValueType> class BufferedReadClientCallback;
+class MTRDataValueDictionaryDecodableType;
 
 @interface MTRReadClientContainer : NSObject
 @property (nonatomic, readwrite) app::ReadClient * readClientPtr;
 @property (nonatomic, readwrite) app::AttributePathParams * pathParams;
 @property (nonatomic, readwrite) app::EventPathParams * eventPathParams;
 @property (nonatomic, readwrite) uint64_t deviceID;
+@property (nonatomic, readwrite) BufferedReadClientCallback<MTRDataValueDictionaryDecodableType> * callbackPtr;
```

Comment:
I don't think this should be needed.  See https://github.com/project-chip/connectedhomeip/pull/25840#discussion_r1168981534

Why is it needed?

Reply:
I commented https://github.com/project-chip/connectedhomeip/pull/25840#discussion_r1169329324